particle/particle/particle.py

```diff
@@ -203,6 +203,80 @@ def all(cls):
 
         return cls._table
 
+    @classmethod
+    def dump_table(cls,
+                   exclusive_fields=[],
+                   exclude_fields=[],
+                   n_rows=-1,
+                   filename=None,
+                   tablefmt='simple'):
+        """
+        Dump the internal particle data CSV table,
+        loading it from the default location if no table has yet been loaded.
+        Optionally dump to a file.
+
+        Note
+        ----
+        Uses the `tabulate` package.
+
+        Parameters
+        ----------
+        exclusive_fields: list, optional, default is []
+            Exclusive list of fields to print out.
+        exclude_fields: list, optional, default is []
+            List of table fields to exclude in the printout.
+            Relevant only when exclusive_fields is not given.
+        n_rows: int, optional, defaults to all rows
+            Number of table rows to print out.
+        filename: str, optional
+            Name of file where to dump the table.
+            By default the table is dumped to stdout.
+        tablefmt: str, optional, default is 'simple'
+            Table formatting option, see the tabulate's package
+            tabulate function for a description of available options.
+            The most common options are:
+            'plain', 'simple', 'grid', 'rst', 'html', 'latex'.
+
+        Examples
+        --------
+        Particle.dump_table()
+        Particle.dump_table(n_rows=5)
+        Particle.dump_table(exclusive_fields=['pdgid', 'pdg_name'])
+        Particle.dump_table(filename='output.txt', tablefmt='rst')
+        """
+        from tabulate import tabulate
+
+        if not cls.table_loaded():
+            cls.load_table()
+
+        # Get all table headers from the class attributes
+        tbl_names = [a.name for a in Particle.__attrs_attrs__]
+
+        if exclusive_fields:
+            tbl_names = exclusive_fields
+        else:
+            for fld in exclude_fields:
+                try:
+                    tbl_names.remove(fld)
+                except:
+                    pass
+
+        # Only dump a given number of rows?
+        tbl_all = cls.all()
+        if n_rows >= 0:
+            tbl_all = tbl_all[:n_rows]
+
+        # Build all table rows
+        tbl = []
+        for p in tbl_all:
+            tbl.append([getattr(p, attr)for attr in tbl_names])
+
+        if filename:
+            with open(filename, 'w') as outfile:
```

Comment:
This should support both file-like objects and filenames.

Reply:
OK, adding a line for this.